File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/HeavyArmor.java
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager;

/* JADX INFO: loaded from: classes3.dex */
public class HeavyArmor extends Quest {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void calculateTargetProgress(int i) {
        this.targetProgress = (long) (Math.pow(1.7d, i - 1) * 500.0d);
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    protected void configure() {
        this.idName = R.string.quest_heavy_armor_name;
        this.idDescription = R.string.quest_heavy_armor_description;
        this.defaultRarity = 1;
        this.minimumDifficulty = 1;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void realignStaticReference() {
        QuestsManager.heavyArmor = this;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/HeavyArmor.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager

class HeavyArmor : Quest() {
    override fun calculateTargetProgress(difficulty: Int) {
        targetProgress = (Math.pow(1.7, (difficulty - 1).toDouble()) * 500.0).toLong()
    }

    override fun configure() {
        idName = R.string.quest_heavy_armor_name
        idDescription = R.string.quest_heavy_armor_description
        defaultRarity = 1
        minimumDifficulty = 1
    }

    override fun realignStaticReference() {
        QuestsManager.heavyArmor = this
    }
}
